src/main/kotlin/com/github/mitchellolsthoorn/testgenie/evosuite/Runner.kt

```diff
@@ -0,0 +1,133 @@
+package com.github.mitchellolsthoorn.testgenie.evosuite
+
+import com.github.mitchellolsthoorn.testgenie.settings.TestGenieSettingsService
+import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.execution.process.*
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.util.Key
+import com.intellij.openapi.util.io.FileUtilRt
+import java.io.File
+import java.nio.charset.Charset
+
+
+class Runner {
+
+    companion object {
+        private val log = Logger.getInstance(Companion::class.java)
+
+        fun runEvoSuiteForClass(projectPath: String, projectClassPath: String, classFQN: String): String {
+
+            val javaPath = "java"// TODO: Source from config
+
+            val pluginsPath = System.getProperty("idea.plugins.path")
+            val evoSuitePath = "$pluginsPath/TestGenie/lib/evosuite.jar"
+
+            val ts = System.currentTimeMillis()
+            val sep = File.separatorChar
+            val testResultDirectory = "${FileUtilRt.getTempDirectory()}${sep}testGenieResults${sep}"
+
+            val testResultName = "test_gen_result_$ts"
+            val serializeResultPath = "\"$testResultDirectory$testResultName\""
+
+            val settingsState = TestGenieSettingsService.getInstance().state
+
```

Comment:
I think that both the test generation for class and method share this logic. Perhaps it can be extracted to remove the duplication?

Reply:
I thought about how exactly I want to do this and decided to leave it for another pull request.